### ISSUE TITLE
Stuff that showed good results for me but is not in master

### ISSUE THREAD

**Issue by Sopel97** [glinscott/nnue-pytorch#130]
- [ ] LR=8.75e-4, gamma=0.992 https://docs.google.com/document/d/1gTlrr02qSNKiXNZ_SuO4-RjK4MXBiFlLE6jvNqqMkAY/edit#heading=h.i7inhg9zufbc - +5 elo at 20k nodes per move
- [ ] L2.6 norm (`loss = torch.pow(torch.abs(p - q), 2.6).mean()`) https://docs.google.com/document/d/1gTlrr02qSNKiXNZ_SuO4-RjK4MXBiFlLE6jvNqqMkAY/edit#heading=h.hb5pfd35aii8 - +10 elo at 20k nodes per move
- [ ] Don't divide LR of the last layer by 10
- [ ] if lambda is used then interpolate before the loss function

**Comment by vondele** [comment]
IMO best would be to produce a net that beats master on fishtest with these settings, and at that point update the trainer.

**Comment by Sopel97** [comment]
I agree, just noting them in a visible place

**Comment by tryingsomestuff** [comment]
May I ask where does the L2.6 norm idea come from ?

**Comment by vondele** [comment]
I experimented with various norms. First a huber loss (which smooth goes from L2 to L1), which turned out to be worse, especially close to L1. The idea that triggered L>2 was the success we had shortly after with augmenting our data with positions for which the net had a large error. Having a L2.6 norm puts more weight on positions with a large error, so is a bit similar in spirit. The precise value of 2.6 was determined with a parameter scan.

**Comment by Torom** [comment]
FYI: `nn-3df94a5762c3.nnue` is a net that was trained with the settings suggested by Sopel. All other parameters were identical to the master network at the time. The settings seem to work well, it could have beaten the then master net.

[STC against former master](https://tests.stockfishchess.org/tests/view/60ccdcbc457376eb8bcab571):
```
LLR: 2.94 (-2.94,2.94) <-0.50,2.50>
Total: 49728 W: 5595 L: 5372 D: 38761
Ptnml(0-2): 257, 4094, 15997, 4201, 315 
```
[LTC against new master](https://tests.stockfishchess.org/tests/view/60cd1a47457376eb8bcab5c6):
```
LLR: -2.93 (-2.94,2.94) <0.50,3.50>
Total: 10736 W: 530 L: 619 D: 9587
Ptnml(0-2): 10, 547, 4350, 444, 17 
```
```
python3 train.py \
    ../data/pylon.binpack \
    ../data/pylon.binpack \
    --gpus 1 \
    --threads 1 \
    --num-workers 1 \
    --batch-size 16384 \
    --progress_bar_refresh_rate 20 \
    --smart-fen-skipping \
    --random-fen-skipping 10 \
    --features=HalfKAv2^ \
    --lambda=1.0 \
    --max_epochs=400 \
    --seed $RANDOM
```
